app/src/main/java/com/ncnf/user/Profile/PublicProfile.java

```diff
@@ -0,0 +1,33 @@
+package com.ncnf.user.Profile;
+
+
+import com.ncnf.user.PrivateUser;
+
+import java.util.concurrent.ExecutionException;
+
+import static com.ncnf.Utils.NAME_KEY;
+
+//TODO : Placeholder for now
+public class PublicProfile implements Profile{
+    private final String username;
+    private final String email;
+
+    public PublicProfile(String username, String email){
+        this.email = email;
+        this.username = username;
+    }
+
+    public PublicProfile(PrivateUser user) throws ExecutionException, InterruptedException {
+        this.email = user.getEmail();
+        this.username = (String) user.getField(NAME_KEY).get().getResult();
```

Comment:
I don't think this respects the separation of concerns. The profile should not query the database, but delegate that job to the CurrentUser or PrivateUser. I would have a controller PublicProfile(name, email, etc).

Reply:
you're right

Reply:
The get() could return null and then the getResult() could make the app crash. Is it well catched outside to be sure to avoid making the app crash for a username ?